### PR TITLE
Added feature to limit number of files to be uploaded

### LINKED ISSUE #615: Limiting the number of files that can be uploaded using config prop
Limit the number of files that can be uploaded using the value given in the config prop for `maxNumberOfFiles`

### PR DESCRIPTION
**Pull request by gaagul**
Fixes #615 

**Description**

- Added: feature to limit the number of files uploaded

**Checklist**

- [x] I have made corresponding changes to the documentation.
- [x] I have added the necessary label (patch/minor/major - If package publish is required).

**Reviewers**

@AbhayVAshokan _a

### CODE REVIEW

**Comment by gaagul** [comment]
@gaagul _a

**Comment by gaagul** [comment]
@AbhayVAshokan _a I have made the utils more concise, Please review the changes.
This [`Video`](https://www.loom.com/share/d3a922df7712455e89e922eb74db619d) explains why I was not able to use uppy's inbuild options.

**Comment by gaagul** [comment]
Note: Since drag and drop behaves differently compared to file-input and adds files directly to uppy object, Removing the files one by one using ids is very complex, hence I have blocked the entire upload in the case of drag and drop and the uppy object will be reset. Please let me know if I have to change this behaviour.

File: lib/components/Attachments/utils.js [hunk 1]
```diff
@@ -1,6 +1,49 @@
...
   mergeRight(DEFAULT_UPPY_CONFIG, { restrictions });
+
+export const selectFiles = ({ previousAttachmentsCount, config, files }) => {
+  const { maxNumberOfFiles } = config;
+
+  if (maxNumberOfFiles) {
+    const remainingAttachments = maxNumberOfFiles - previousAttachmentsCount;
+
+    if (remainingAttachments <= 0) {
+      Toastr.warning("You can't add more files");
```
Comment (AbhayVAshokan) [inline lib/components/Attachments/utils.js:16]: Make the error precise. It should be the same as max files error.

**Review by AbhayVAshokan — APPROVED** [review]
LGTM. @gaagul _a please merge after this small change.

**Pull request merged**